omc3/scripts/luminosity_imbalance.py

```diff
@@ -0,0 +1,118 @@
+from math import sqrt
+
+import tfs
+
+from omc3.utils.logging_tools import get_logger, list2str
+from generic_parser import entrypoint, EntryPointParameters
+from generic_parser.entrypoint_parser import save_options_to_config
+
+LOG = get_logger(__name__)
+
+
+def get_params():
+    return EntryPointParameters(
+            tfs=dict(
+                required=True,
+                type=str,
+                help="TFS file containing the β* and error for "
+                     "B{1,2}{H,V}IP{1,5}"
+            ),
+            inplace=dict(
+                required=False,
+                action='store_true',
+                help="Add the luminosity imbalance in the source TFS file as a"
+                     "header"
+            )
+    )
+
+
+def _validate_tfs(df):
+    # First validate the labels, which correspond to the IP and Beam
+    expected_labels = set(('IP1B1', 'IP1B2', 'IP5B1', 'IP5B2'))
+    if set(df['LABEL']) != expected_labels:
+        LOG.error("Expected labels in the TFS file not found. Expected labels:"
+                  " {}".format(expected_labels))
+        raise KeyError('Labels not found in TFS file')
+
+    # Validate the columns we need: BETSTAR{X,Y} and ERRBETSTAR{X,Y}
+    expected_columns = ('BETSTARX', 'BETSTARY', 'ERRBETSTARX', 'ERRBETSTARY')
+    if not all([column in df.columns for column in expected_columns]):
+        LOG.error("Expected columns in the TFS file not found. Expected "
+                  "columns: {}".format(expected_columns))
+        raise KeyError('Columns not found in TFS file')
+
+
+def get_imbalance(df):
+    # Get the individual rows for the IP1 and IP5
+    ip1 = df.loc[df['LABEL'].str.startswith('IP1')]
```

Comment:
Loop over ips? -> half the code

Reply:
I prefer improved readability over number of lines

Reply:
ok, then:

Loop over ips? -> more readable code as less line repetitions